app/models/user.rb

```diff
@@ -1,5 +1,20 @@
 class User < ActiveRecord::Base
 	has_secure_password
 
-	validates :user_name, presence: true, uniqueness: true
+	validates :user_name, presence: true
+
+	def self.find_or_create_from_auth(data)
```

Comment:
Assignment Branch Condition size for find_or_create_from_auth is too high. [22.02/15]<br>Tab detected.